fix: fix flakiness in test HealthCheckConfigValidator#startValidation…

hofi1 opened the pull request:
…sShouldFailIfAHealthCheckConfiguredButNotRegistered

###### Problem:
<!-- Explain the context and why you're making that change. What is the problem you're trying to solve? In some cases there is not a problem and this can be thought of being the motivation for your change. -->

###### Solution:
<!-- Describe the modifications you've done. -->

###### Result:
<!-- What will change as a result of your pull request? Note that sometimes this section is unnecessary because it is self-explanatory based on the solution. -->

Sujishark commented:
Using the contains() method repeatedly becomes inefficient as the number of values to check grows. Can you check on the alternative data structures, apart from 'Set', that you can suggest to preserve the order of elements ?

ThugJudy commented:
Using the contains() method for every element of the list does not seem scaleable. Consider using another data structure that maintains order. It would be nice if you could add the nondex command used to reproduce the issue.

hofi1 commented:
Hey @Sujishark and @ThugJudy ,
thanks for your good feedback, I appreciate it!
I updated the original comment and added the command to reproduce the bug.
Furthermore, I changed the two `.contains` calls to only use one.

I don't think it is right to force the order of the values in the string, because the ordering just doesn't matter.
But I am open for further suggestions and a discussion on that.

zzjas commented:
Maybe you also want to assert that the message length. The test should fail if the message contains 2,3,4.

It's your call to spend more time addressing the comments or not, but we approve you to proceed to open a real PR. Once you open a real PR, please mark this tentative PR as `Opened` in your `tentative_pr.csv` file and also raise a PR to IDoFT marking this as `Opened`. Thanks!
